app/helpers/avatar_helper.rb

```diff
@@ -1,44 +1,43 @@
+# frozen_string_literal: true
+
 module AvatarHelper
+  DEFAULT_PROFILE_IMAGE = ActionController::Base.helpers.image_url('profile')
+  DEFAULT_PROFILE_IMAGE_SIZE = 80
+
+  # Gravatar will use a default image if one is not found. Having gravatar serve
+  # the default image is not desired. Instead force an error by using a bad
+  # default url and let our fallback image code take effect.
   def avatar_url(user, options = {})
-    if user.nil?
-      image_path('profile.png')
-    else
-      gravatar_id = Digest::MD5.hexdigest(user.email.downcase)
-      size = options.fetch(:size, 64).to_i
-      "https://secure.gravatar.com/avatar/#{gravatar_id}.png?r=PG&s=#{size}"
-    end
+    return DEFAULT_PROFILE_IMAGE if user.nil? || !user.email.include?('@')
+
+    gravatar_id = Digest::MD5.hexdigest(user.email.downcase)
+    size = options.fetch(:size, DEFAULT_PROFILE_IMAGE_SIZE).to_i * 2 # Retna displays mean dot density can be higher.
+    "https://secure.gravatar.com/avatar/#{gravatar_id}.png?r=PG&s=#{size}&d=forceErrorOnDefault"
```

Comment:
can you explain this option? I'm looking for it in https://en.gravatar.com/site/implement/images/ but can't find it

Reply:
If no default is provided gravatar will provide *their* default image, the gravatar logo. We don't want that. Our code specifies that if a network error occurs while loading (air-gapped instance) we use our fallback images. But if no error occurs, and the user doesn't have gravatar we want to ensure our fallback image is still applied. This triggers our fallback usage by purposefully supplying a url that is inaccessible.

Based on the conversation today about air-gapped instances. We could also provide a URL to the default asset on the VM, and if it works it works, if it doesn't work we *might* still fallback properly, otherwise it won't work and it sounds like that's okay.